src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -54,14 +54,33 @@ protected function render_button(Component\Component $component, RendererInterfa
 			$tpl->setVariable("DISABLED", self::DISABLED_CLASS);
 		}
 
+		$this->maybe_render_id($component, $tpl);
+
 		return $tpl->get();
 	}
 
-	protected function render_close() {
+	protected function render_close($component) {
 		$tpl = $this->getTemplate("tpl.close.html", false, false);
+		$this->maybe_render_id($component, $tpl);
 		return $tpl->get();
 	}
 
+	protected function maybe_render_id($component, $tpl) {
+		$id = $this->bindJavaScript($component);
+		if ($id !== null) {
+			$tpl->setVariable("ID", "id=\"$id\"");
+		}
+		else {
+			// This is required for the close button. I tried to use the purge_unfilled_vars
+			// option for getTemplate in render_close, but it doesn't do what i expect. I
+			// checked the passing of the parameter down to HTML_Template_IT, which also
+			// works. There seems to be some arcane spell or dance, that HTML_Template_IT
+			// want me to perform, that i don't know.
+			// -- klees, 2016-08-17
+			$tpl->setVariable("ID", "");
```

Comment:
I don't know if i understand your proposal correctly, but how would that work? You would still need to introduce an ID-placeholder in your template, right? To me it seemed to be more transparent to leave the templating stuff to the implementor of the Renderer completely, thus stop at the generation of the id. It also might be required to put ids in more than one place in a rendered component, e.g. to link a modal to its trigger (or something similar). I'm not too sure about all that, so i would be glad for clarification and opinions.